File: frontend/src/models/Deck.ts
export enum DeckVisibility {
  Public = 'Public',
  Private = 'Private',
}

export interface Deck {
  deckId: number;
  userId: string;
  title: string;
  description: string;
  language: string;
  visibility: string;
  coverImageBlobName: string | null;
  joinCode: string;
  createdAt: string;
  updatedAt: string;
}

export interface DeckRequest {
  title: string;
  description?: string;
  language: string;
  visibility: DeckVisibility;
  coverImageBlobName?: string;
}

File: backend/noava/noava/Models/Deck.cs
using noava.Models.Enums;

namespace noava.Models
{
    public class Deck
    {
        public int DeckId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string Language { get; set; } = string.Empty;
        public DeckVisibility Visibility { get; set; } = DeckVisibility.Private;
        public string? CoverImageBlobName { get; set; }
        public string UserId { get; set; } = string.Empty;
        public string JoinCode { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public ICollection<DeckUser> DeckUsers { get; set; } = new List<DeckUser>();
        public ICollection<ClassroomDeck> ClassroomDecks { get; set; } = new List<ClassroomDeck>();
    }
}